### ISSUE TITLE
Bug: Unexpected YAML Merge Key expansion when changing format and value of Anchored child nodes

### ISSUE THREAD

**Issue by hemnstill** [wwkimball/yamlpath#191]
## Operating System

1. Name/Distribution: Windows
2. Version: 10

## Version of Python and packages in use at the time of the issue.

1. [Distribution](https://wiki.python.org/moin/PythonDistributions): CPython
2. Python Version: 3.x
3. Version of yamlpath installed: 3.6.7
4. Version of ruamel.yaml installed: 0.17.21

## Minimum sample of YAML (or compatible) data necessary to trigger the issue

test_commands_yaml_set.py

```
    def test_update_base_value(self, script_runner, tmp_path_factory):
        yamlin = """---
common: &base
  TEST_COMMON_SETTING: 'a1'
TestService:
  <<: *base
"""
        yamlout = """---
common: &base
  TEST_COMMON_SETTING: a2
TestService:
  <<: *base
  TEST_COMMON_SETTING: a2
"""
        yaml_file = create_temp_yaml_file(tmp_path_factory, yamlin)
        result = script_runner.run(
            self.command,
            "--change=common.TEST_COMMON_SETTING",
            "--value=a2",
            yaml_file
        )
        assert result.success, result.stderr

        with open(yaml_file, 'r') as fhnd:
            filedat = fhnd.read()
        assert filedat == yamlout
```

## Expected Outcome
```
---
common: &base
  TEST_COMMON_SETTING: a2
TestService:
  <<: *base
```

if you change `TEST_COMMON_SETTING: 'a1'` -> `TEST_COMMON_SETTING: a1` then everything is ok.

looks like bug in check:
```
def recurse(data, parent, parentref, reference_node, replacement_node):
...
    if (hasattr(val, "anchor"):
```
value with type `SingelQuotedScalarString` has an attr `anchor`.

**Comment by wwkimball** [comment]
I'm confused by the way you wrote a test-case for this (only the input YAML, what actually happens, and what should happen are asked for)... that isn't a test that exists anywhere in the project code, so I have to assume you're offering to to be added.  In so doing, you appear to WANT the command to create an extra node under `TestService` when changing `common`.  But then your Expected Outcome doesn't show the extra node, contradicting the test case you wrote, so... I'm confused.

I created an in.yaml and ran the command I can see in your test-case.  Doing this shows the behavior I expect:

```
$ python --version
Python 3.9.13

$ pip list | grep ruamel.yaml
ruamel.yaml            0.17.21
ruamel.yaml.clib       0.2.6

$ yaml-set --version
yaml-set 3.6.7

$ cat in.yaml
---
common: &base
  TEST_COMMON_SETTING: 'a1'
TestService:
  <<: *base

$ cat in.yaml | yaml-set --change='common.TEST_COMMON_SETTING' --value=a2
---
common: &base
  TEST_COMMON_SETTING: a2
TestService:
  <<: *base
  TEST_COMMON_SETTING: a2
```

The value of common.TEST_COMMON_SETTING correctly changes from `'a1'` (where the string demarcation symbols are not required by YAML) to `a2` (having correctly removed the unnecessary string demarcation symbols).

If you wish to preserve the unnecessary string demarcation symbols, also specify `--format`, as this:

```
$ cat in.yaml | yaml-set --change='common.TEST_COMMON_SETTING' --value=a2 --format=squote
---
common: &base
  TEST_COMMON_SETTING: 'a2'
TestService:
  <<: *base
  TEST_COMMON_SETTING: 'a2'
```

The statement that there is a "bug in check" is incorrect.  The code is behaving correctly.

**Comment by wwkimball** [comment]
This issue is fixed in the 3.6.8 release, which is now available.
